### ISSUE TITLE
No music since v1.2

### ISSUE THREAD

**Issue by Vic2077** [alexbatalov/fallout2-ce#251]
Since v1.2 music is not playing anymore (plays fine in older builds). Same for Fallout1-ce (v1.0 doesn't play backgroun music).
Main menu and videos have music, everything else (like background noises like dog barks in Arroyo) is working, but the location music doesn't play.

**Comment by APAmk2** [comment]
There is some reports about that, but strange, in F2, F:N and F:S sound is fine, everything plays fine

**Comment by Vic2077** [comment]
Hmm, maybe I should check Sonora and Nevada too. But then again, the only thing that changed is the apk file. Game files are the same, and with older apks music still plays fine with the same game files.

**Comment by QcSamuel** [comment]
tried the solution for ios on fallout1-ce and music is still not playing

**Comment by Vic2077** [comment]
> tried the solution on fallout1-ce repo for ios and music is still not playing

Also tried to rename all files and folders and lines in the cfg file to CAPITALS - no result. (although I've tried this on fallout1-ce, but in Fallout 2 it all already is in lowercase letters and music still doesn't play.)

**Comment by bencarr1337** [comment]
Tried renaming all the files in fallout2.cfg but still no music for me either

**Comment by bencarr1337** [comment]
Hello All, i have found the issue i was having. In the "data/sounds/music" folder, all of the music files were lower case, if you change them to upper case it works fine. 

Here is a python script to do it if you're lazy like me. Put it in the music folder and run it:

`import os

# get the current directory
current_directory = os.getcwd()

# loop through all the files in the directory
for filename in os.listdir(current_directory):
    # check if the filename is a file (not a directory)
    if os.path.isfile(os.path.join(current_directory, filename)):
        # create the new filename by converting it to uppercase
        new_filename = filename.upper()
        # rename the file
        os.rename(os.path.join(current_directory, filename), os.path.join(current_directory, new_filename))
`

**Comment by APAmk2** [comment]
That is not reason of problem, this is too stupid thing that need to test as firstly as can, we are tried that already, nothing changed

**Comment by Vic2077** [comment]
> works fine

not for me

**Comment by bencarr1337** [comment]
Yeah, it only worked for me on the trials level. The world map, Klamath etc it still doesn’t work for me.

would really like a resolution for this as it kills the atmosphere.

**Comment by alexbatalov** [comment]
I've checked last debug version, music works as expected (checked main menu, arroyo, temple of trails, den). If it doesn't work, please post screenshots from Files app showing your folder hierarchy (data/sound/music) and contents of your `fallout2.cfg`.

**Comment by alexbatalov** [comment]
Any additional info on this one?

**Comment by bencarr1337** [comment]
> Any additional info on this one?

I extracted the file that contains the references to the music files (I forget what it’s called now) and it looks like the filenames it refers to are a mix of uppercase and lowercase so some places work and some don’t depending on your config file. Obviously this doesn’t make a difference on windows but on nix OS’s it does.

**Comment by alexbatalov** [comment]
Check if #214 helps (https://github.com/alexbatalov/fallout2-ce/actions/runs/4670923559).

**Comment by alexbatalov** [comment]
That's strange. Can you check v1.1 without changing anything in the config/directory layout so we can be sure it's something in the code?

**Comment by drjfaust** [comment]
I did some of my own testing on the issue both in the tagged (v1.2.0) and current-commit (c8d4585) versions:

1. Without my patch, or file renaming, there indeed appears to be no music.
2. Interestingly, fallout2-ce now appears to sometimes attempt to load empty-string ("\0") filenames, which I found out after applying my own patch. That definitely didn't use to happen.
3. After I applied an additional check at the end of compat_windows_path_to_native() (i.e. put an "if (d)" before "closedir(d)", the game launched, and there was music at the main menu (which I remember being the first problem with music I ever noticed before making the patch), but attempting to start a new game, or loading a saved game leads to segfault.

So, I think most likely some bad filenames are being supplied to compat_windows_path_to_native(), and presumably somewhere else later.

**Comment by alexbatalov** [comment]
Let's even things:
- Download latest debug build of the main branch (https://github.com/alexbatalov/fallout2-ce/actions/runs/4670076031).
- Grab HQ music (https://github.com/BGforgeNet/Fallout2-HQ-music/releases/tag/v8)
- In your Fallout 2 directory (that's where `fallout2.cfg` is) create `sound/music/` (lowercased), unpack HQ music into this folder.
- Uppercase all `acm` files (i.e. music in main menu `07DESERT.ACM`).
- Delete `fallout2.cfg` (defaults are fine for this setup).

Post your results.

**Comment by QcSamuel** [comment]
> Let's even things:
> 
> * Download latest debug build of the main branch (https://github.com/alexbatalov/fallout2-ce/actions/runs/4670076031).
> * Grab HQ music (https://github.com/BGforgeNet/Fallout2-HQ-music/releases/tag/v8)
> * In your Fallout 2 directory (that's where `fallout2.cfg` is) create `sound/music/` (lowercased), unpack HQ music into this folder.
> * Uppercase all `acm` files (i.e. music in main menu `07DESERT.ACM`).
> * Delete `fallout2.cfg` (defaults are fine for this setup).
> 
> Post your results.
tried it on ios and the main menu music play for 2 second because stopping and no music in game

**Comment by alexbatalov** [comment]
@Vic2077 @QcSamuel Guys can you check it one more time with the latest debug build?

**Comment by QcSamuel** [comment]
@alexbatalov tried the latest debug it works now for me on ios

**Comment by 007hacky007** [comment]
@alexbatalov I can confirm music works just fine in 1.1 release and does work only for few seconds on 1.2 release (macOS build). Same behavior with HQ music as well as default music files included in the GOG's release. All tests with music files renamed to uppercase. 

I'm unable to test latest build, because [last week's artifacts](https://github.com/alexbatalov/fallout2-ce/actions/runs/6340300373) already expired.

**Comment by alexbatalov** [comment]
That's odd, last time I played on a Mac music was there. Will check and get back with result in a couple of days.

**Comment by alexbatalov** [comment]
I've tried v1.2 from releases as well as building from source both debug and release configurations, music works on Intel Mac. Post your `fallout2.cfg` and make a screenshot of your directory layout (tree view, be sure to reveal `acm` files). I'll try to replicate that one to one to see what's going on. Thanks.

**Comment by 007hacky007** [comment]
I've got Apple Silicon - M1 Pro to be exact. I'm running macOS Ventura 13.6. 

My `fallout2.cfg` is completely default. I've even deleted the file and let it to be recreated with default values as you suggested in one of your previous replies. Anyway I'm attaching both tree output and `fallout2.cfg` file. 

Note: `sound/music2` directory is the original .acm music files that came with GOG release (just renamed to uppercase) and `sound/music` is the [HQ music](https://github.com/BGforgeNet/Fallout2-HQ-music/releases/tag/v8) renamed to uppercase. 

With 1.2 release the music works for few seconds and then it stops. And just with replacing the .app with 1.1 release, it starts working (without any additional steps - i.e. changing the `fallout2.cfg`. 

[treeFallout2.txt](https://github.com/alexbatalov/fallout2-ce/files/12844577/treeFallout2.txt)
[fallout2.cfg.txt](https://github.com/alexbatalov/fallout2-ce/files/12844581/fallout2.cfg.txt)

**Comment by alexbatalov** [comment]
Can you please check [latest debug build](https://github.com/alexbatalov/fallout2-ce/actions/runs/6460068828) just to make it has not been already fixed.

**Comment by 007hacky007** [comment]
I can confirm that music works just fine with the latest debug build.

**Comment by rumly111** [comment]
for me, I had to rename the files to `blah.ACM` to make it work , so only the extension is uppercase. I'm on version 1.2.0

```
    for a in * ; do mv $a `basename $a .acm`.ACM -v ; done
```

**Comment by ArminiusTux** [comment]
Greetings, I must also confirm that this is still an issue with the latest code base.

Quick fix for Windows:
```
cmd
cd \fallout2-folder\sound\music
ren *.acm *.ACM
```

**Comment by alexbatalov** [comment]
@ArminiusTux Hey, can you please check the latest debug build? You've said you are on WinARM64, is your file system case-sensitive?

**Comment by ArminiusTux** [comment]
Dear @alexbatalov , Windows on ARM is as any other Windows variant case-insensitive (i.e. README.TXT can not coexist with readme.txt in the same folder).
Although you can apparently [change that now](https://learn.microsoft.com/en-us/windows/wsl/case-sensitivity) on a folder by folder case.

As for your [latest build linked above](https://github.com/alexbatalov/fallout2-ce/issues/251#issuecomment-1753440388), the file downloads seem to have expired:
![image](https://github.com/alexbatalov/fallout2-ce/assets/44533868/bf491886-d6db-4b2d-9db8-838d6b7e3467)

**Comment by alexbatalov** [comment]
Please check this one: https://github.com/alexbatalov/fallout2-ce/actions/runs/8581579710

**Comment by tonurics** [comment]
I was also able to get music working on Linux:

1. Download the HQ Music: https://github.com/BGforgeNet/Fallout2-HQ-music/releases/tag/v8
2. Extract into: ./sound/music/ (move or remove old music files)
3. Change the file extensions to upper case: `rename '.acm' '.ACM' *`

I'm guessing the important bit is the file extension case. But included all steps to allow someone else to reproduce my results.

**Comment by alexbatalov** [comment]
Resolved in v1.3

**Comment by zlice** [comment]
i don't think 1.3 is fixed. someone else on windows 11 just had to do the acm -> ACM rename with https://github.com/alexbatalov/fallout2-ce/releases/download/v1.3.0/fallout2-ce-windows-x86.zip
